### ISSUE TITLE
Propagate Claims to PAM Modules

### ISSUE THREAD

**Issue by russjones** [gravitational/teleport#5902]
At the moment, Teleport propagates the Teleport username, logins, and roles to PAM modules through environment variables. Customers have asked for way to claims to PAM modules.

**Comment by russjones** [comment]
Application Access is doing something similar by updating file configuration to specify which headers should be propagated. https://github.com/gravitational/teleport/issues/4762

**Comment by russjones** [comment]
@r0mant What do you think?

**Comment by r0mant** [comment]
@russjones Do you mean something like

```yaml
ssh_service:
  pam:
    enabled: true
    service_name: "sshd"
    use_pam_auth: true
    env:
      USER_EMAIL: "{{external.email}}"
      USER_GROUPS: "{{external.groups}}"
```

?

**Comment by xacrimon** [comment]
Yeah, also wondering about this. Also what claims have to be supported? Looking at the application access issue there's some discussion around things like `Impersonate-User` do we need to support such things?

**Comment by xacrimon** [comment]
@russjones

**Comment by russjones** [comment]
@xacrimon No need to support `Impersonate-User`. Implementation will be similar to what we have for variables within roles right now, you might even be able to re-use a lot of the code. But instead let's make it a resource instead so PAM configuration can be applied cluster wide.

Sent you a calendar invite, let's sync about it tomorrow.

**Comment by xacrimon** [comment]
Fixed by #6158
